### PR TITLE
chore(RHINENG-11732): Rework SP filters to use ORM

### PR DESCRIPTION
**Pull request by kruai**
# Overview

This PR is being created to address [RHINENG-11732](https://issues.redhat.com/browse/RHINENG-11732).
We originally implemented the System Profile filters in #1656 by generating SQL strings for text filters. This PR re-implements these filters to use SQLAlchemy's ORM filter functions instead, which is both more secure and easier to read.

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [x] Descriptive comments provided in complex code blocks
- [ ] Include raw query examples in the PR description, if adding/modifying SQL query
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [x] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [x] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: api/filtering/filtering_common.py [hunk 1]
```diff
@@ -1,26 +1,38 @@
...
-    "nil": "IS NULL",
-    "not_nil": "IS NOT NULL",
+    "lt": ColumnOperators.__lt__,
+    "lte": ColumnOperators.__le__,
+    "gt": ColumnOperators.__gt__,
+    "gte": ColumnOperators.__ge__,
+    "eq": ColumnOperators.__eq__,
+    "neq": ColumnOperators.__ne__,
+    "is": ColumnOperators.is_,
+    "contains": "contains",
```
Comment (jpramos123) [inline api/filtering/filtering_common.py:14]: Maybe use `ColumnOperators.contains`?
```suggestion
    "contains": ColumnOperators.contains,
```

File: api/filtering/filtering_common.py [hunk 2]
```diff
@@ -1,26 +1,38 @@
...
-    "integer": "=",
-    "date-time": "=",
-    "array": "?",
+    "string": ColumnOperators.__eq__,
+    "wildcard": ColumnOperators.ilike,
+    "boolean": ColumnOperators.is_,
+    "operating_system": ColumnOperators.__eq__,
+    "integer": ColumnOperators.__eq__,
+    "date-time": ColumnOperators.__eq__,
+    "array": "contains",
```
Comment (jpramos123) [inline api/filtering/filtering_common.py:34]: Maybe use `ColumnOperators.contains`?
```suggestion
    "array": ColumnOperators.contains,
```

**Review by jpramos123 — COMMENTED** [review]
One suggestion, please see if it applies.

https://docs.sqlalchemy.org/en/20/core/sqlelement.html#sqlalchemy.sql.expression.ColumnOperators.contains

Thanks

File: api/filtering/filtering_common.py (hunk 1 above)
Comment (kruai) [reply api/filtering/filtering_common.py:14]: There actually isn't a ColumnOperator that checks for whether an array contains a value. `ColumnOperators.contains` checks for a substring within a string

File: api/filtering/filtering_common.py (hunk 2 above)
Comment (kruai) [reply api/filtering/filtering_common.py:34]: same as above

**Review by jpramos123 — DISMISSED** [review]

**Comment by fstavela** [comment]
@kruai The tests are failing because when I filter hosts by some boolean field, HBI always returns only hosts with that field being `True`, regardless of what value I provide in the filter. I tried these values in the filter: true, True, TRUE, false, False, FALSE

**Comment by kruai** [comment]
Looks like there's an issue combining nil/not_nil with arrays; working on a fix

**Review by jpramos123 — APPROVED** [review]

**Review by thearifismail — APPROVED** [review]
LGTM

**Pull request merged**